phpBB/includes/acp/acp_board.php

```diff
@@ -909,6 +912,119 @@ function board_disable($value, $key)
 	}
 
 	/**
+	* Default settings for the WYSIWYG editors
+	*/
+	function wysiwyg_converters($submit)
+	{
+		global $config, $request, $user, $phpbb_container, $phpbb_dispatcher, $template;
+
+		$new_wysiwyg = $current_wysiwyg = $config['wysiwyg_type'];
+		$new_default_modes = $current_default_modes = unserialize($config['wysiwyg_default_default_mode']);
+		$new_button_modes = $current_button_modes = unserialize($config['wysiwyg_default_buttons_mode']);
+
+		$editor_data = array();
+
+		foreach ($phpbb_container->get('wysiwyg.converters_collection') as $name => $converter)
+		{
+			$available_modes = $converter->get_available_modes();
+			$button_modes = $converter->get_available_button_modes();
+
+			$editor_data[$name] = array(
+				'EDITOR_ID' => $name,
+				'EDITOR_TITLE' => $converter->get_name(),
+
+				'WYSIWYG_MODES_AVAILABLE' => $available_modes,
+				'WYSIWYG_BUTTON_MODES_AVAILABLE' => $button_modes,
+
+				'S_WYSIWYG_MODE' => $current_default_modes[$name],
+				'S_WYSIWYG_BUTTON_MODE' => $current_button_modes[$name],
+			);
+
+		}
+
+		if ($submit)
+		{
+			$set_data = $request->variable('wysiwyg', array(), false, \phpbb\request\request_interface::POST);
+			$editor_new_data = $editor_data;
+
+			var_dump($set_data);
+
+			if (isset($editor_data[$set_data['default_editor']]))
+			{
+				$new_wysiwyg = $set_data['default_editor'];
+			}
+
+			foreach ($set_data as $name => $new_data)
+			{
+				if ($editor_data[$name]['WYSIWYG_MODES_AVAILABLE'] & $new_data['default_mode'])
+				{
+					$new_default_modes[$name]							= $new_data['default_mode'];
+					$editor_new_data[$name]['S_WYSIWYG_MODE']			= $new_data['default_mode'];
+				}
+				if ($editor_data[$name]['WYSIWYG_BUTTON_MODES_AVAILABLE'] & $new_data['default_button_mode'])
+				{
+					$new_button_modes[$name]							= $new_data['default_button_mode'];
```

Comment:
I'm unsure if this is a right way of doing it because the same data is in multiple places.
I'll think about it later.